### PR TITLE
[System.Windows.Forms] Locking Window Handle List

### PR DESCRIPTION
**Pull request by mlancione**
A lock to the EventQueue Window Handle List is required to prevent multiple threads from modifying the list;

### CODE REVIEW

File: mcs/class/System.Windows.Forms/System.Windows.Forms/XEventQueue.cs [hunk 1]
```diff
@@ -142,16 +142,30 @@ public class PaintQueue {
 			}
 
 			public int Count {
-				get { return hwnds.Count; }
+				get
+				{
+					int Count = 0;
+					lock ( hwnds )
+					{
+						Count = hwnds.Count;
```
Comment (marek-safar) [inline mcs/class/System.Windows.Forms/System.Windows.Forms/XEventQueue.cs]: It's ok to just return hwnds.Count here

**Comment by marek-safar** [comment]
Please following existing coding style

**Comment by mlancione** [comment]
The coding style changes have been made.

**Comment by ludovic-henry** [comment]
approve

**Comment by ludovic-henry** [comment]
We get a `System.Windows.Forms/XEventQueue.cs(185,10): error CS0103: The name 'wnd' does not exist in the current context` compilation error

**Comment by mlancione** [comment]
Compile error fixed.

**Pull request merged**